core/http/client/src/main/java/org/eclipse/rdf4j/http/client/RDF4JProtocolSession.java

```diff
@@ -620,11 +623,14 @@ public synchronized void beginTransaction(IsolationLevel isolationLevel)
 			method.setHeader("Content-Type", Protocol.FORM_MIME_TYPE + "; charset=utf-8");
 
 			List<NameValuePair> params = new ArrayList<>();
-			if (isolationLevel != null) {
+
+			for (TransactionSetting transactionSetting : transactionSettings) {
+				if (transactionSetting == null)
+					continue;
 				params.add(
 						new BasicNameValuePair(
-								Protocol.ISOLATION_LEVEL_PARAM_NAME,
-								isolationLevel.getURI().stringValue()
+								TRANSACTION_SETTINGS_PREFIX + transactionSetting.getName(),
+								transactionSetting.getValue()
 						)
```

Comment:
This is not backward compatible. The REST API _requires_ a transaction isolation level to be sent, but this opens the option of not supplying one. It also removes the actual Isolation level parameter that is part of the protocol, instead encoding everything as a generic transaction setting. 

I'll push some fixes for this.

Reply:
Looks like I misremembered that it's required at the protocol level. We'll still need to keep the specific parameter around though.